### ISSUE TITLE
Returning $_.Exception removes the reason the exception was thrown in Invoke-AkamaiRestMethod

### ISSUE THREAD

**Issue by brantpeery** [akamai/akamaipowershell#27]
When Invoke-AkamaiRestMethod throws an error, the detailed reason of the web call failure (the underlying Invoke-RestMethod failure body) is stored in the $_.ErrorDetails object. When the below code is run, there is no way to tell why the call may have failed (if it failed for example because their was no more licenses for enrollments)
```powershell
Invoke-AkamaiRestMethod -Method GET -Path "$($enrollment.changes[0])/input/info/third-party-csr" -EdgeRCFile ..\TestData\edgerc -AdditionalHeaders @{Accept="application/vnd.akamai.cps.csr.v1+json"} 
```
Returning the entire error object allows the calling functions to determine why the call failed and respond gracefully to the failure.
* The code for Invoke-AkamaiRestMethod for PS v5.1 already returns the whole error object $_
* The code for Invoke-AkamaiRestMethod for PS v6+ returns only the $_.Exception which will hide the ErrorDetail from the caller
* The code in some of the Akamai specific functions like New-CPSEnrollment also filters the error to just $\_.Exception which can confuse the caller as for example 404 not found is used for both a non-existent location and a contract without the correct resources or all the licenses have already been used. This detail is contained in the $_.ErrorDetails object.

**Comment by brantpeery** [comment]
On further investigation, I have found that the Akamai API sends the reason for http error responses in the body of the response. Invoke-RestMethod puts the response.body in the ErrorDetail.Message property. So the error detail looks like this on an error from Akamai:

```JSON
{
  "type" : "https://akab-asd-kzl7zfdgovchetzf.luna.akamaiapis.net/cps/v2/error-types/dns-name-validation-failure",
  "title" : "DNS names failed validation.",
  "instance" : "https://akab-asdf-kzl7zfdgovchetzf.luna.akamaiapis.net/cps/v2/error-types/dns-name-validation-failure?id=f4f58c40bf5841d0ae596178d1db6718",
  "detail" : "DNS names failed validation."
}
```
This information is not presented anywhere in the Exception object. The only place to get it is through the $_.ErrorDetail property. As you can see, this is VERY important information that will help the developer or end user know what to do next, after an exception is encountered. 
I looked through the code, and it looks like the standard from all the scripts that make a call to Invoke-AkamaiRestMethod catch and rethrow, but do so with 
``` powershell
    catch {
        throw $_.Exception
    }  
```
This will remove the extra information and the resulting error object will not have an ErrorDetail property. I propose that ALL rethrows of exceptions coming from Invoke-AkamaiRestMethod be changed to 
```powershell
    catch {
        throw $_
    }  
```
-or- 
The invoking method deal with the separate reasons that the API  call would have failed and give a new exception object that clearly states the error reason. This shouldn't effect any existing code as the resulting $_ will have the same structure as the current code, except it will retain some of the metadata from the original error.

**Comment by stuartio** [comment]
Hi @brantpeery we have released an update for this in 1.8.0. Can you check your issue is resolved please?
